parser.go

```diff
@@ -26,7 +26,8 @@ import (
 EscapedBackslash = "\\"
 EscapedDollar    = ( "\$" | "$$")
 Identifier       = letter { letters | digit | "_" }
-Expansion        = "$" ( Identifier | Brace )
+EscapedExpansion = EscapedDollar { Identifier }
+Expansion        = { "$" ( Identifier | Brace ) | EscapedExpansion }
```

Comment:
i need to update this change to the BNF, it's from a previous iteration of this change